.github/workflows/build-plugin-archive.yml

```diff
@@ -0,0 +1,241 @@
+name: Create plugin archive
+
+on:
+  workflow_call:
+    inputs:
+      PLUGIN_MAIN_FILE:
+        description: The path/name of the plugin main file.
+        required: false
+        default: './index.php'
+        type: string
+      PLUGIN_VERSION:
+        description: The plugin version.
+        required: true
+        type: string
+      PLUGIN_BASENAME:
+        description: The base name of the resulting zip archive. Falls back to the repository name.
+        default: ''
+        required: false
+        type: string
+      COMPOSER_ARGS_DEV:
+        description: Set of arguments passed to Composer for build setup.
+        default: '--prefer-dist'
+        required: false
+        type: string
+      COMPOSER_ARGS:
+        description: Set of arguments passed to Composer for production setup.
+        default: '--no-dev --no-scripts --prefer-dist --optimize-autoloader'
+        required: false
+        type: string
+      NPM_REGISTRY_DOMAIN:
+        description: Domain of the private npm registry.
+        default: https://npm.pkg.github.com/
+        required: false
+        type: string
+      NODE_VERSION:
+        description: Node version to use for asset compilation.
+        default: 16
+        required: false
+        type: string
+      PHP_VERSION:
+        description: PHP version to use during packaging.
+        default: 8.2
+        required: false
+        type: string
+      EXTRA_PHP_FILE:
+        description: Path to a custom php script to run before creating the release archive.
+        default: ''
+        required: false
+        type: string
+      EXTRA_PHP:
+        description: Run custom php code before creating the release archive.
+        default: ''
+        required: false
+        type: string
+    secrets:
+      COMPOSER_AUTH_JSON:
+        description: Authentication for privately hosted packages and repositories as a JSON formatted object.
+        required: false
+      NPM_REGISTRY_TOKEN:
+        description: Authentication for the private npm registry.
+        required: false
+      ENV_VARS:
+        description: Additional environment variables as a JSON formatted object.
+        required: false
+
+jobs:
+  create-plugin-archive:
+    timeout-minutes: 5
+    runs-on: ubuntu-latest
+    env:
+      # Disables symlinking of local path repositories.
+      # During development, symlinking is preferrable.
+      # In resulting builds, you will likely want to ship the actual install location and remove the source directory.
+      COMPOSER_MIRROR_PATH_REPOS: 1
+      COMPOSER_AUTH: '${{ secrets.COMPOSER_AUTH_JSON }}'
+      NODE_AUTH_TOKEN: ${{ secrets.NPM_REGISTRY_TOKEN }}
+    outputs:
+      artifact: ${{ steps.set-artifact-name.outputs.artifact }}
+    steps:
+
+      #
+      # Project setup
+      #
+
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup package base name
```

Comment:
Yeah. Initially I wanted to pretend this could work for themes as well, which is why I used "package".

I've since dropped that idea, but didn't update the wording everywhere